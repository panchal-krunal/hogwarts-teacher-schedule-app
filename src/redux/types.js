export const TOGGLE_TEACHER_PRESENCE = 'TOGGLE_TEACHER_PRESENCE'
export const ALLOCATE_SUBJECT = 'ALLOCATE_SUBJECT'
export const ASSIGN_TEACHER = 'ASSIGN_TEACHER'