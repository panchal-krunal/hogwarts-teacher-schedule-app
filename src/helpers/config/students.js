const students = {
  BVK8aC82Y8: {
    name: "Harry Potter",
    subject: "gaJoeo1e1K",
    teacherAllocated: "e6Xah9MirI",
  },
  ubUFqkgphE: {
    name: "Hermione Granger",
    subject: "gaJoeo1e1K",
    teacherAllocated: "",
  },
  SJPbyBV7Fs: {
    name: "Ron Weasley",
    subject: "gaJoeo1e1K",
    teacherAllocated: "SV4mio1jYY",
  },
  RXpXxanmzB: {
    name: "Draco Malfoy",
    subject: "gaJoeo1e1K",
    teacherAllocated: "e6Xah9MirI",
  },
  aLAfQu2hbf: {
    name: "Padma Patil",
    subject: "gaJoeo1e1K",
    teacherAllocated: "",
  },
  MYLGVVThfF: {
    name: "Luna Lovegood",
    subject: "gaJoeo1e1K",
    teacherAllocated: "SV4mio1jYY",
  },
};

export default students;