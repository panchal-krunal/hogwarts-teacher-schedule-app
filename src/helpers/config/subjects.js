const subjects = {
  gaJoeo1e1K: {
    name: "Potions master",
  },
  ibuUcWMKR5: {
    name: "Defense Against the Dark Arts",
  },
};
export default subjects;