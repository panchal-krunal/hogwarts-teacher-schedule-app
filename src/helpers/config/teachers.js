const teachers = {
  ql_HP3WWb5: {
    name: "Professor Dumbledore",
    subject: "",
    reporting_to: null,
    designation: "Headmaster",
    isPresent: true,
  },
  ihUHzDMgDf: {
    name: "Minerva McGonagall",
    subject: "",
    reporting_to: "ql_HP3WWb5",
    designation: "Headmistress",
    isPresent: true,
  },
  vWljuJ10kr: {
    name: "Rubeus Hagrid",
    subject: "gaJoeo1e1K",
    reporting_to: "ihUHzDMgDf",
    designation: "Standby Professor",
    isPresent: true,
    isHeadingSubject:true
  },
  e6Xah9MirI: {
    name: "Horace Slughorn",
    subject: "gaJoeo1e1K",
    reporting_to: "vWljuJ10kr",
    designation: "",
    isPresent: true,
  },
  SV4mio1jYY: {
    name: "Severus Snape",
    subject: "gaJoeo1e1K",
    reporting_to: "vWljuJ10kr",
    designation: "",
    isPresent: true,
  },
  xhxpdg8RzN: {
    name: "Alastor Moody",
    subject: "ibuUcWMKR5",
    reporting_to: "ihUHzDMgDf",
    designation: "Standby Professor",
    isPresent: true,
    isHeadingSubject:true
  },
  rCkLspfCxE: {
    name: "Remus Lupin",
    subject: "ibuUcWMKR5",
    reporting_to: "xhxpdg8RzN",
    designation: "",
    isPresent: true,
  },
  rCkLspfCxE: {
    name: "Gilderoy Lockhart",
    subject: "ibuUcWMKR5",
    reporting_to: "xhxpdg8RzN",
    designation: "",
    isPresent: true,
  },
};

export default teachers;
