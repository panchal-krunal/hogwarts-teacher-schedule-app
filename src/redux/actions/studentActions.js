import { ASSIGN_TEACHER } from "../types";

export const assignTeacher = (data) => (dispatch) => {
  dispatch({
    type: ASSIGN_TEACHER,
    payload: data,
  });
};
