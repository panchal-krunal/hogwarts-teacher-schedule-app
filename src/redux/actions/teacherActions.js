import { TOGGLE_TEACHER_PRESENCE } from "../types";

export const toggleAttendance = (id, isPresent) => (dispatch) => {
  dispatch({
    type: TOGGLE_TEACHER_PRESENCE,
    payload: {
      id,
      isPresent,
    },
  });
};
