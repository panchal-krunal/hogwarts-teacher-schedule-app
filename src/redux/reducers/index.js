import { combineReducers } from "redux";
import teacherReducer from "./teacherReducer";
import subjectReducer from "./subjectReducer";
import studentReducer from "./studentReducer";
const rootReducer = combineReducers({
  teacherReducer,
  subjectReducer,
  studentReducer
});

export default rootReducer;
