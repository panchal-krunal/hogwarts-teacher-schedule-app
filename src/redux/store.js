import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import reducer from "./reducers";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

const persistConfig = {
  key: "root",
  storage,
};

const persistedReducer = persistReducer(persistConfig, reducer);

const store = createStore(
  persistedReducer,
  compose(
    applyMiddleware(thunk),
    window.devToolsExtension ? window.devToolsExtension() : (f) => f
  )
);

const persistor = persistStore(store);

export { store, persistor };
