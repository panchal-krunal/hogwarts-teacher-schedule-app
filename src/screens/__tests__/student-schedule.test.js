import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import configureStore from "redux-mock-store";
import teachers from "../../helpers/config/teachers";
import subjects from "../../helpers/config/subjects";
import * as ReactReduxHooks from "../../helpers/react-redux-hooks";
// import * as teacherActions from "../../redux/actions/teacherActions";
import thunk from "redux-thunk";
import StudendSchedule from "../student-schedule";
import students from "../../helpers/config/students";
import { assignTeacher } from "../../redux/actions/studentActions";
import { ASSIGN_TEACHER } from "../../redux/types";
Enzyme.configure({ adapter: new Adapter() });

describe("teacher attendance component renders correctly", () => {
  let wrapper;
  let useEffect;
  let store;
  let updatedStudentObj;
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const mockUseEffect = () => {
    useEffect.mockImplementationOnce((f) => f());
  };
  beforeEach(() => {
    store = mockStore({
      teacherReducer: {
        teachers: { ...teachers },
      },
      subjectReducer: {
        subjects: { ...subjects },
      },
      studentReducer: {
        students: null,
      },
    });
    useEffect = jest.spyOn(React, "useEffect");
    mockUseEffect(); // important to do it twice
    mockUseEffect();
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation((state) => store.getState());

    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);

    wrapper = mount(<StudendSchedule store={store} />);

    updatedStudentObj = [
      {
        student: "Harry Potter",
        subject: "gaJoeo1e1K",
        teacher: "Professor Dumbledore",
      },
      {
        student: "Hermione Granger",
        subject: "gaJoeo1e1K",
        teacher: "Professor Dumbledore",
      },
      {
        student: "Ron Weasley",
        subject: "gaJoeo1e1K",
        teacher: "Professor Dumbledore",
      },
      {
        student: "Draco Malfoy",
        subject: "gaJoeo1e1K",
        teacher: "Professor Dumbledore",
      },
      {
        student: "Padma Patil",
        subject: "gaJoeo1e1K",
        teacher: "Professor Dumbledore",
      },
      {
        student: "Luna Lovegood",
        subject: "gaJoeo1e1K",
        teacher: "Professor Dumbledore",
      },
    ];
  });

  it("should find table", () => {
    const tableElement = wrapper.find("table");
    expect(tableElement).toHaveLength(1);
  });
  it("should dispatch assign teacher action", () => {
    store.dispatch(assignTeacher(updatedStudentObj));
    const action = store.getActions();
    const expectedAction = [
      { type: ASSIGN_TEACHER, payload: updatedStudentObj },
    ];
    expect(action).toEqual(expectedAction);
  });
  it("should dispatch assign teacher action and update store values", () => {
    store.dispatch(assignTeacher(updatedStudentObj));
    store = mockStore({
      teacherReducer: {
        teachers: { ...teachers },
      },
      subjectReducer: {
        subjects: { ...subjects },
      },
      studentReducer: {
        students: updatedStudentObj,
      },
    });
    wrapper = mount(<StudendSchedule store={store} />);
    const tableElement= wrapper.find('table')
    const rowElement= tableElement.find('tr')
    expect(rowElement).toHaveLength(7)
  });
});
