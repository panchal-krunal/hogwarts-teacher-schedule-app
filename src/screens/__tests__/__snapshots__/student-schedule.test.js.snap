// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teacher attendance component renders correctly should render correctly 1`] = `
<StudendSchedule
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Container
    fluid={false}
  >
    <div
      className="container"
    >
      <ForwardRef
        bordered={true}
        hover={true}
        striped={true}
      >
        <table
          className="table table-striped table-bordered table-hover"
        >
          <thead>
            <tr>
              <th>
                #
              </th>
              <th>
                Student Name
              </th>
              <th>
                Subject
              </th>
              <th>
                Allocated Teacher
              </th>
            </tr>
          </thead>
        </table>
      </ForwardRef>
    </div>
  </Container>
</StudendSchedule>
`;
